### PR TITLE
jabba: Install shell integration scripts

### LINKED ISSUE #138: Jabba installed through homebrew not working
running the command sequence:

```
brew install jabba
jabba install zulu@1.8.121
jabba use zulu@1.8.121
echo $JAVA_HOME
```

results in the `$JAVA_HOME` environment variable not being set.

Also running `jabba current` doesn't return anything either.

`zulu@1.8.121` is listed as being installed when I run `jabba ls`

I can confirm these same command work as expected when installed through the `curl` script.

Jabba version 0.9.2
OSX High Sierra

### PR DESCRIPTION
**Pull request by kabel**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Closes https://github.com/shyiko/jabba/issues/138

### CODE REVIEW

**Comment by SMillerDev** [comment]
If upstream doesn't ship this we shouldn't either

**Comment by kabel** [comment]
They do ship it, but we don't install via their `install.sh` script that _would_ handle it. I'm not sure why the current install implementation uses `go` rather than any of the documented installation steps, but it likely deals with the `install.sh` writing these shells scripts outside of the Homebrew sandbox. That is, `install.sh` writes to `$HOME/.jabba`

**Comment by SMillerDev** [comment]
Their `install.sh` doesn't compile from source, it downloads binaries so it's not possible to use that.

**Comment by kabel** [comment]
I see that. Upon further review of their release tarball, they have a `Makefile` that does build from source an install via `install.sh` (with some environment variables disabling the binary download and the shell integration script generation). If you that that's a better path forward, I can take a stab at refactoring the `def install` to use `make` and do some replacements/patching to the `install.sh` to get things to install in the Cellar vs `$HOME`.

**Comment by SMillerDev** [comment]
Yeah, that sounds good. We just want to make sure it builds from source, the commands to get there don't matter much.

**Comment by kabel** [comment]
Latest push does the necessary path fixes with `inreplace` and then uses `make`. This should generate the shell integration scripts.

**Comment by kabel** [comment]
Since this is now tracking a new upstream repo/fork, I've suggested some changes upstream to make packaging easier (without the need to do patching). https://github.com/Jabba-Team/jabba/pull/17

File: Formula/jabba.rb [hunk 1]
```diff
@@ -22,14 +23,39 @@ class Jabba < Formula
...
+      s.gsub! "${JABBA_HOME}/bin", bin.to_s
+      s.gsub! "${JABBA_HOME}/jabba.sh", "#{pkgshare}/jabba.sh"
+      s.gsub! "${JABBA_HOME}/jabba.fish", "#{pkgshare}/jabba.fish"
+    end
+
+    pkgshare.mkpath
+
     dir = buildpath/"src/github.com/Jabba-Team/jabba"
     dir.install buildpath.children
     cd dir do
```
Comment (SMillerDev) [inline Formula/jabba.rb:44]: Does it work without this?

Comment (kabel) [reply Formula/jabba.rb]: It isn't, once we take out the `go` environment overrides.

**Comment by kabel** [comment]
Updated the test to confirm that shell integration is working through the FD 3 of `jabba use`

File: Formula/jabba.rb [hunk 2]
```diff
@@ -20,25 +21,58 @@ class Jabba < Formula
...
+    inreplace "install.sh" do |s|
+      # Call unaliased rm
+      s.gsub! "  rm -f", "  command rm -f"
+
+      # Homebrew install locations
+      s.gsub! "$JABBA_HOME_TO_EXPORT/bin/jabba", "#{opt_bin}/jabba"
+      s.gsub! "${JABBA_HOME}/bin", bin.to_s
+      s.gsub! "${JABBA_HOME}/jabba.sh", "#{pkgshare}/jabba.sh"
+      s.gsub! "${JABBA_HOME}/jabba.fish", "#{pkgshare}/jabba.fish"
     end
```
Comment (SMillerDev) [inline Formula/jabba.rb:36]: Could you make an upstream issue or pull request to make these configurable and reference it here?

Comment (kabel) [reply Formula/jabba.rb]: Added at https://github.com/Homebrew/homebrew-core/pull/119957/files#diff-21b44b2361fc470b49d829a6c3d512881cf16415102a6230531a2ee6c56f6294R27

File: Formula/jabba.rb [hunk 3]
```diff
@@ -20,25 +21,57 @@ class Jabba < Formula
...
+    fd3.rewind
+    fd3.each do |line|
+      m = java_home_pattern.match line
+      next if m.nil?
+
+      jdk_path = m[1]
+    end
+    fd3.unlink
+
+    refute_equal "", jdk_path
```
Comment (SMillerDev) [inline Formula/jabba.rb:73]: how is all this different than using this?
```rb
assert_match /^export JAVA_HOME="([^"]+)"/, shell_output("#{bin}/jabba use #{jdk_version}")
```

Comment (kabel) [reply Formula/jabba.rb:73]: It doesn't go to output, so `shell_output` would be empty. It uses a specific file descriptor used by the generated shell script integration to hide the shell commands (setting environment variables) the will be executed in the background. For simplicity sake, I guess we could add `3>&1` to the command to redirect that special stream to the normally empty output of that command.

Comment (kabel) [reply Formula/jabba.rb]: Simplified in latest push.

**Review by SMillerDev — APPROVED** [review]

**Comment by kabel** [comment]
Can the `do not merge` label be removed and then merged?

**Comment by kabel** [comment]
@p-linnane Bump. Can we get this merged?

**Pull request closed without merging**